lib/ansible/modules/cloud/ovirt/ovirt_templates.py

```diff
@@ -324,34 +324,34 @@ def main():
             templates_service = storage_domain_service.templates_service()
 
             # Find the the unregistered Template we want to register:
-            templates = templates_service.list(unregistered=True)
+            templates = templates_service.list(query={'unregistered': 'true'})
             template = next(
-                (t for t in templates if t.name == module.params['name']),
+                (t for t in templates if t.id == module.params['id']),
                 None
             )
             changed = False
             if template is None:
-                # Test if template is registered:
-                template = templates_module.search_entity()
-                if template is None:
+                template_service = templates_service.template_service(module.params['id'])
+                try:
+                    template = template_service.get()
+                # TODO: TypeError is being thrown from _check_content_type, although it should be SDK.error
+                except TypeError:
                     raise ValueError(
-                        "Template with name '%s' wasn't found." % module.params['name']
+                        "Template with id '%s' wasn't found." % (module.params['id'])
                     )
             else:
+                # Register the template into the system:
                 changed = True
                 template_service = templates_service.template_service(template.id)
-                # Register the template into the system:
                 template_service.register(
                     cluster=otypes.Cluster(
                         name=module.params['cluster']
-                    ) if module.params['cluster'] else None,
-                    template=otypes.Template(
-                        name=module.params['name'],
-                    ),
+                    ) if module.params['cluster'] else None
                 )
-                if module.params['wait']:
-                    template = wait_for_import(module, templates_service)
 
+                # Register template is merely a DB operation so there is no reason to wait.
```

Comment:
yes, there are no tasks, so when the engine command is ended we are sure that the Template is in the DB (or not if encountered an error), I will re-verify that this is the behavior again just to be sure.
Besides that, regarding the search of the entity, I encountered an issue here that the search for the entity is filtered with name, and since the entity has only id, I think that the search function should be changed